### PR TITLE
Make strided_index_range an aggregate

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

File: submdspan/submdspan.md [hunk 1]
```diff
@@ -417,25 +417,21 @@ array<IndexType, sizeof...(SliceSpecifiers)> @_src-indicies_@(const array<IndexT
...
+  ExtentType extent;
+  StrideType stride;
 };
 ```
 
+[1]{.pnum} `strided_index_range` is an aggregate type.
+
+[2]{.pnum} *Mandates:*
+
+  * `OffsetType`, `ExtentType`, and `StrideType` are signed or unsigned integer types, or are specializations of `integral_constant` that are not a specializaiton of  `bool_constant`.
```
Comment (mhoemmen) [inline submdspan/submdspan.md:434]:
```suggestion
  * `OffsetType`, `ExtentType`, and `StrideType` are signed or unsigned integer types, or are specializations of `integral_constant` that are not a specialization of  `bool_constant`.
```

File: submdspan/submdspan.md [hunk 2]
```diff
@@ -417,25 +417,21 @@ array<IndexType, sizeof...(SliceSpecifiers)> @_src-indicies_@(const array<IndexT
...
-  OffsetType _@_offset_@; // exposition only
-  ExtentType _@_extent_@; // exposition only
-  StrideType _@_stride_@; // exposition only
+  OffsetType offset;
+  ExtentType extent;
+  StrideType stride;
 };
 ```
 
+[1]{.pnum} `strided_index_range` is an aggregate type.
```
Comment (mhoemmen) [inline submdspan/submdspan.md:430]: I was thinking of being even more insistent here.

`strided_index_range` is an aggregate **[dcl.init.aggr]**.
It is trivially copyable **[class.prop]**
if `OffsetType`, `ExtentType`, and `StrideType` are all trivially copyable.
It is a structural type **[temp.param]**
if `OffsetType`, `ExtentType`, and `StrideType` are all structural types.

**Review by mhoemmen — APPROVED** [review]
I was working on this in parallel, but most of my changes are non-wording.  I'll submit those separately.  Thanks!

File: submdspan/submdspan.md (hunk 2 above)
Comment (crtrott) [reply submdspan/submdspan.md:430]: isn't that all implied?

**Pull request merged**